services/authoring/src/repositories/apps-repository.ts

```diff
@@ -33,12 +40,89 @@ export default class AppsRepository {
     return this.apps;
   }
 
-  async saveApp(appId, manifest, author) {
-    return await this._gitTransactionManager.write(async repo => {
+  async createApp(appId: string, manifest: AppManifest, author) {
+    return await this._gitTransactionManager.write(async (repo) => {
       await repo.updateFile(`external_apps/${appId}.json`, JSON.stringify(manifest, null, 4));
       const commitId = await repo.commitAndPush(`created app ${appId}`, author);
       await this.refresh();
       return commitId;
     });
   }
+
+  async updateApp(appId: string, manifest: Partial<Pick<AppManifest, 'name' | 'permissions'>>, author) {
+    const appFileName = `external_apps/${appId}.json`;
+    return await this._gitTransactionManager.write(async (repo) => {
+      const app = this.apps[appId];
+      if (!app) {
+        throw new Error(`The app with id ${appId} doesn't exist`);
+      }
+      const updatedApp = { ...app, ...manifest };
+      await repo.updateFile(appFileName, JSON.stringify(updatedApp));
+      const commitId = await repo.commitAndPush(`updated app ${appId}`, author);
+      await this.refresh();
+      return commitId;
+    });
+  }
+
+  async deleteApp(appId: string, author) {
+    return await this._gitTransactionManager.write(async (repo) => {
+      await repo.deleteFile(`external_apps/${appId}.json`);
+      const commitId = await repo.commitAndPush(`deleted app ${appId}`, author);
+      await this.refresh();
+      return commitId;
+    });
+  }
+
+  async createSecretKey(appId: string, keyId: string, secretKey: AppSecretKey, author) {
+    const appFileName = `external_apps/${appId}.json`;
+    return await this._gitTransactionManager.write(async (repo) => {
+      const app = this.apps[appId];
+      if (!app) {
+        throw new Error(`The app with id ${appId} doesn't exist`);
+      }
+      // Support legacy apps without secretKeyMap
+      if (!app.secretKeyMap) {
+        app.secretKeyMap = app.secretKeys.reduce((acc, cur, idx) => ({ ...acc, [`legacy-${idx}`]: cur }), {});
```

Comment:
maybe change legacy to "default"

Reply:
theoretically we can have several keys so it's a bit weird to call all of them default..